pyspecProcScripts/third_level/process_IR.py

```diff
@@ -16,21 +19,55 @@ def select_pathway(s,pathway):
 def as_scan_nbr(d):
     '''since we need to relabel vd frequently- we make a method'''
     return d.C.setaxis('vd','#').set_units('vd','scan #')
-
-def process_IR(s, label='', fl=None,
-        this_l = 0.032,
+# this is written like you're trying to run it as a script!
+signal_pathway = {'ph1':0,'ph2':1}
+excluded_pathways = [(0,0)]
+def process_IR(s, this_l = 0.032,
         l = sqrt(np.logspace(-8.0,0.5,35)),
-        signal_pathway = {'ph1':0,'ph2':1},
-        excluded_pathways = [(0,0)],
         clock_correction = True,
         W=6.2,
         f_range = (None,None),
         t_range = (None,83e-3),
         IR = True,
         flip=False,
-        sign = None,
-        ILT=False):
-    s *= sign
+        sgn = None,
+        ILT=False,
+        fl=None):
+    """Fully processes inversion recovery experiments to produce a fit inversion
+    recovery curve to extrapolate the T1 time from.
+
+    Parameters
+    ==========
+    s:      nddata
+    this_l: int
+            knee of the L curve that gives best balance between signal and
+    l:      ndarray
+            spacing for the L curve minimization
+    clock_correction:   boolean
+                        especially needed at times in spincore to correct for 
+                        drift in frequency and time.
+    W:      int
+            repetition delay set for FIR experiments
+    f_range:    tuple
+                range in which the signal resides in the frequency domain.
+    t_range:    tuple
+                range in which the signal resides in the time domain.
+    IR:     boolean
+            True for a true inversion recovery fitting, false for FIR fitting.
+    flip:   boolean 
+            At higher powers it's been noted the signal can flip signs sometimes
+            inverting the recovery curve-this corrects for it
+    sign:   nddata  
+            the nddata with the collected signs of the data along the signal that
+            upon multiplication will convert all signal to the same sign. Useful
+            for datasets where a null is present.
+    ILT:    boolean
+            If true, will produce an ILT image plot for visualizing the T1 as a function
+            of offset.
+    fl:     boolean
+            option to show figures produced by processing script.
+        """        
+    s *= sgn
```

Comment:
why is this even part of this pull request?

Reply:
you want to save this in your own branch, but this function as a whole will likely be superseded by the table of integrals function.

Reply:
need to test alignment on actual IR data to ensure it works as it should